src/main/java/com/elastic/support/diagnostics/commands/UserRoleCheckCmd.java

```diff
@@ -0,0 +1,85 @@
+package com.elastic.support.diagnostics.commands;
+
+import com.elastic.support.config.Version;
+import com.elastic.support.diagnostics.chain.Command;
+import com.elastic.support.diagnostics.chain.DiagnosticContext;
+import com.elastic.support.rest.RestClient;
+import com.elastic.support.rest.RestResult;
+import com.elastic.support.util.JsonYamlUtils;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserRoleCheckCmd implements Command {
+
+    Logger logger = LogManager.getLogger(UserRoleCheckCmd.class);
+
+    @Override
+    public void execute(DiagnosticContext context) {
+
+        RestClient restClient = context.getEsRestClient();
+        String user = context.getDiagnosticInputs().getUser();
+
+        Version version = context.getVersion();
+        String url = null;
+        int major = version.getMajor();
+        int minor = version.getMinor();
+
+        if(major == 2){
+            url = "/_shield/user/" + user;
+        }
+        else if (major <= 6){
```

Comment:
This will be used for version `1`, which is never going to return anything useful for the given request since X-Pack and its APIs wasn't a thing until 5.0. Perhaps we should just skip this for `1.x` and log a warning saying it's unverified (and perhaps a tip to upgrade)?